### PR TITLE
Gate menu toolbar and gate creation 

### PR DESCRIPTION
**Pull request by BartoszMichalak13**
Adding basing menu with choice of logic gates and ability to drag and drop to create new gate on main viewport

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -294,6 +297,97 @@ static void render_scene(Scene& scene, Vec2 const viewport_size)
...
+  ImGui::EndChild();
+  ImGui::End();
+}
+
+Gate_Kind button_id_to_gate_kind() {
+  switch (last_button_id)
+  {
+  case 0:
+      return Gate_Kind::e_and;
+      LOG_INFO("ADDING AND GATE");
```
Comment (kociap) [inline src/main.cpp]: why yell?

File: src/main.cpp [hunk 2]
```diff
@@ -407,11 +501,24 @@ int main(int argc, char* argv[])
     // Close the dock window.
     ImGui::End();
 
+    create_menu_window();
+
+
```
Comment (kociap) [inline src/main.cpp:521]: superfluous whitespace

File: src/main.cpp [hunk 3]
```diff
@@ -294,6 +297,97 @@ static void render_scene(Scene& scene, Vec2 const viewport_size)
...
+    // Menu bar goes here...
+    ImGui::EndMenuBar();
+  }
+
+  // for each of the toolbar buttons
+  for(int i = 0; i < 7; ++i) {
+    /*
+    Contains possible button names in men
+    */
+    char* button_list[7] = {"AND",\
```
Comment (kociap) [inline src/main.cpp]: why line extensions?

File: src/main.cpp [hunk 4]
```diff
@@ -294,6 +297,97 @@ static void render_scene(Scene& scene, Vec2 const viewport_size)
...
+{
+  ImGui::Begin("Window with menu bar and button toolbar", nullptr,
+               ImGuiWindowFlags_MenuBar);
+  if(ImGui::BeginMenuBar()) {
+    // Menu bar goes here...
+    ImGui::EndMenuBar();
+  }
+
+  // for each of the toolbar buttons
+  for(int i = 0; i < 7; ++i) {
```
Comment (kociap) [inline src/main.cpp]: what if there are 8 kind?

**Review by kociap — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 1 above)
Comment (BartoszMichalak13) [reply src/main.cpp]: Changed to lowercase letters

File: src/main.cpp (hunk 2 above)
Comment (BartoszMichalak13) [reply src/main.cpp:521]: Changed to just 1 newline

File: src/main.cpp (hunk 3 above)
Comment (BartoszMichalak13) [reply src/main.cpp]: Changed to one-line comment

File: src/main.cpp (hunk 4 above)
Comment (BartoszMichalak13) [reply src/main.cpp]: What are possible solutions? One would be defining X-macro with gate-types and creating names based on it, but this macro would have to be updated each time we add new gate type (now we need to update loop length and add new name for new gate) so it is somewhat comparable. 

Is X-macro the way or what do you suggest?

Comment (BartoszMichalak13) [reply src/main.cpp]: Conditions met (to my knowledge at least)

**Pull request closed without merging**